fission/src/ui/modals/mirabuf/ImportMirabufModal.tsx

```diff
@@ -3,80 +3,142 @@ import Modal, { ModalPropsImpl } from "@/components/Modal"
         >
             <Label className="text-wrap break-all">{name}</Label>
             <Button value={buttonText} onClick={onClick} />
-        </div>
+            {secondaryButtonText && secondaryOnClick && <Button value={secondaryButtonText} onClick={secondaryOnClick} />} </div>
     )
 }
 
+export type MiraManifest = {
+    robots: string[]
+    fields: string[]
+}
+
 const ImportMirabufModal: React.FC<ModalPropsImpl> = ({ modalId }) => {
     // update tooltip based on type of drivetrain, receive message from Synthesis
     // const { showTooltip } = useTooltipControlContext()
 
     const [selectedHub, setSelectedHub] = useState<Hub | undefined>(undefined)
-    const [selectedProject, setSelectedProject] = useState<Project | undefined>(undefined)
-    const [selectedFolder, setSelectedFolder] = useState<Folder | undefined>(undefined)
+    const [selectedProject, setSelectedProject] = useState<Project | undefined>()
+    const [selectedFolder, setSelectedFolder] = useState<Folder | undefined>()
+    const [manifest, setManifest] = useState<MiraManifest | undefined>()
+    const [files, setFiles] = useState<Data[] | undefined>()
+    const auth = APS.getAuth();
 
-    const [hubs, setHubs] = useState<Hub[] | undefined>(undefined)
-    // prettier-ignore
     useEffect(() => {
-        (async () => {
-            setHubs(await getHubs())
-        })()
+        if (auth) {
+            getHubs().then(async (hubs) => {
+                if (!hubs) return;
+                const fileData = []
+                for (const hub of hubs) {
+                    const projects = await getProjects(hub)
+                    if (!projects) continue;
+                    for (const project of projects) {
+                        const data = await searchRootForMira(project)
+                        if (data)
+                            fileData.push(...data)
+                    }
+                }
+                setFiles(fileData)
+            })
+        }
     }, [])
 
-    const [projects, setProjects] = useState<Project[] | undefined>(undefined)
-    // prettier-ignore
-    useEffect(() => {
-        (async () => {
-            if (selectedHub) {
-                setProjects(await getProjects(selectedHub))
-            }
-        })()
-    }, [selectedHub])
+    const cachedRobots: MirabufCacheInfo[] = Object.values(MirabufCachingService.GetCacheMap(MiraType.ROBOT))
+    const cachedFields: MirabufCacheInfo[] = Object.values(MirabufCachingService.GetCacheMap(MiraType.FIELD))
 
-    const [folderData, setFolderData] = useState<Data[] | undefined>(undefined)
-    // prettier-ignore
     useEffect(() => {
-        (async () => {
-            if (selectedProject) {
-                console.log("Project has been selected")
-                if (selectedFolder) {
-                    console.log(`Selecting folder '${selectedFolder.displayName}'`)
+        fetch('/api/mira/manifest.json').then(x => x.json()).then(data => {
```

Comment:
Could you utilize the `import.meta.env.VITE_SYNTHESIS_SERVER` variable to adjust where this manifest file is retrieved from?